backend/src/packages/article-reactions/article-reaction.repository.ts

```diff
@@ -0,0 +1,112 @@
+import { type IRepository } from '~/libs/interfaces/interfaces.js';
+
+import { ArticleReactionEntity } from './article-reaction.entity.js';
+import { type ArticleReactionModel } from './article-reaction.model.js';
+import { getReactionsQuery } from './libs/helpers/helpers.js';
+import { type ArticleReactionCreateResponseDto } from './libs/types/types.js';
+
+class ArticleReactionRepository implements IRepository {
+  private articleReactionModel: typeof ArticleReactionModel;
+
+  public constructor(articleReactionModel: typeof ArticleReactionModel) {
+    this.articleReactionModel = articleReactionModel;
+  }
+
+  public findAll(): ReturnType<IRepository['findAll']> {
+    return Promise.resolve([]);
+  }
+
+  public async find(id: number): Promise<ArticleReactionEntity | null> {
+    const reaction = await this.articleReactionModel
+      .query()
+      .findById(id)
+      .execute();
+
+    if (!reaction) {
+      return null;
+    }
+
+    return ArticleReactionEntity.initialize(reaction);
+  }
+
+  public async findAllByArticleId(
+    articleId: number,
+  ): Promise<ArticleReactionCreateResponseDto> {
+    const reactions = await this.articleReactionModel
+      .query()
+      .select(
+        getReactionsQuery(this.articleReactionModel)(true),
+        getReactionsQuery(this.articleReactionModel)(false),
+      )
+      .where('articleId', articleId)
+      .first()
+      .castTo<ArticleReactionModel & ArticleReactionCreateResponseDto>()
+      .execute();
+
+    if (!reactions) {
+      return {
+        likeCount: 0,
+        dislikeCount: 0,
+      };
+    }
+
+    return reactions;
+  }
+
+  public async findArticleReaction(
+    userId: number,
+    articleId: number,
+  ): Promise<ArticleReactionEntity | null> {
+    const reaction = await this.articleReactionModel
+      .query()
+      .where('userId', userId)
+      .andWhere('articleId', articleId)
+      .first()
+      .execute();
+
+    if (!reaction) {
+      return null;
+    }
+
+    return ArticleReactionEntity.initialize(reaction);
+  }
+
+  public async create(
+    entity: ArticleReactionEntity,
+  ): Promise<ArticleReactionEntity> {
+    const { isLike, userId, articleId } = entity.toNewObject();
+
+    const reaction = await this.articleReactionModel
+      .query()
+      .insert({
+        isLike,
+        userId,
+        articleId,
+      })
+      .returning('*')
+      .execute();
+
+    return ArticleReactionEntity.initialize(reaction);
+  }
+
+  public async update(
+    entity: ArticleReactionEntity,
+  ): Promise<ArticleReactionEntity> {
+    const { id, ...payload } = entity.toObject();
+
+    const updatedReaction = await this.articleReactionModel
+      .query()
+      .patchAndFetchById(id, payload)
+      .execute();
+
+    return ArticleReactionEntity.initialize(updatedReaction);
+  }
+
+  public async delete(id: number): Promise<boolean> {
+    return Boolean(
+      await this.articleReactionModel.query().deleteById(id).execute(),
+    );
```

Comment:
Return deleted record
```suggestion
    return this.articleReactionModel.query().deleteById(id).returning('*').execute();
```

Reply:
Implemented returning a deleted record, but the code is slightly different from the suggestion - [613b3c3](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/104/commits/613b3c3a25512880cce5dd8479d6b4933d6f1f6a)